tdvt/tdvt/metadata/priority.csv

```diff
@@ -0,0 +1,327 @@
+date.datepart.dayofyear	Required
+date.datepart.hour	Required
+date.datepart.iso-quarter	Recommended
+date.datepart.iso-week	Recommended
+date.datepart.iso-weekday	Recommended
+date.datepart.iso-year	Recommended
+date.datepart.minute	Required
+date.datepart.month	Required
+date.datepart.nulls	Recommended
+date.datepart.quarter	Required
+date.datepart.second	Required
+date.datepart.second.ms	Required
+date.datepart.sow.day	Required
+date.datepart.sow.dayofyear	Required
+date.datepart.sow.hour	Required
+date.datepart.sow.minute	Required
+date.datepart.sow.month	Required
+date.datepart.sow.nulls	Recommended
+date.datepart.sow.quarter	Required
+date.datepart.sow.second	Required
+date.datepart.sow.week	Required
+date.datepart.sow.weekday	Required
+date.datepart.sow.year	Required
+date.datepart.week	Required
+date.datepart.weekday	Required
+date.datepart.year	Required
+date.datetrunc.day	Required
+date.datetrunc.dayofyear	Required
+date.datetrunc.hour	Required
+date.datetrunc.iso-quarter	Recommended
+date.datetrunc.iso-week	Recommended
+date.datetrunc.iso-weekday	Recommended
+date.datetrunc.iso-year	Recommended
+date.datetrunc.minute	Required
+date.datetrunc.month	Required
+date.datetrunc.nulls	Recommended
+date.datetrunc.quarter	Required
+date.datetrunc.second	Required
+date.datetrunc.sow.day	Required
+date.datetrunc.sow.dayofyear	Required
+date.datetrunc.sow.hour	Required
+date.datetrunc.sow.minute	Required
+date.datetrunc.sow.month	Required
+date.datetrunc.sow.nulls	Recommended
+date.datetrunc.sow.quarter	Required
+date.datetrunc.sow.second	Required
+date.datetrunc.sow.week	Required
+date.datetrunc.sow.weekday	Required
+date.datetrunc.sow.year	Required
+date.datetrunc.week	Required
+date.datetrunc.weekday	Required
+date.datetrunc.year	Required
+date.math	Recommended
+date.math.date_minus_date	Recommended
+date.max	Required
+date.min	Required
+date.misc	Required
+date.now	Recommended
+date.today	Recommended
+datetime.cast.date	Required
+datetime.cast.float	Required
+datetime.cast.str	Required
+Filter.Add_to_context	Required
+Filter.Date_In	Required
+Filter.datetime_fractional	Recommended
+Filter.Exclude-list-not-null	Required
+Filter.Exclude-range-not-null	Required
+Filter.FilterBy-to-no-results	Required
+Filter.Q_date_instance_filter	Required
+Filter.Q_date_month_instance_filter	Required
+Filter.QFilter-max-NotNull	Required
+Filter.QFilter-max-Null	Required
+Filter.QFilter-min-NotNull	Required
+Filter.QFilter-min-Null	Required
+Filter.QFilter-minmax-NotNull	Required
+Filter.QFilter-minmax-Null	Required
+Filter.QFilter-NotNull	Required
+Filter.QFilter-Null	Required
+Filter.slicing_Q_date_instance_filter	Required
+Filter.slicing_Q_date_month_instance_filter	Required
+Filter.TopN-context	Required
+Filter.Trademark	Smoke
+join.null.bool	Recommended
+join.null.date	Recommended
+join.null.datetime	Recommended
+join.null.int	Recommended
+join.null.real	Recommended
+join.null.str	Recommended
+lod.1_Difference From with Cross Join	Required for Relational
```

Comment:
I think at the moment we only support relational databases, there could be an argument for making these fully required to avoid confusion (if we only ever intend to support relational and not other types of database)

Reply:
By "relational", I mean actual ACID-compliant relational databases, not hadoop or NoSQL databases hiding behind a SQL engine. But I should change this to "Required if supported" if it makes that a little more clear.